static/app/views/settings/components/forms/selectField.tsx

```diff
@@ -87,33 +98,35 @@ export default class SelectField<
         {...otherProps}
         alignRight={small}
         field={({onChange, onBlur, required: _required, ...props}) => (
-          <Confirm
-            renderMessage={({selectedValue}) =>
-              confirm && selectedValue
-                ? confirm[selectedValue.value]
-                : // Set a default confirm message
-                  t('Continue with these changes?')
-            }
-            onCancel={() => {}}
-            onConfirm={this.handleChange.bind(this, onBlur, onChange)}
-          >
-            {({open}) => (
-              <SelectControl
-                {...props}
-                clearable={allowClear}
-                multiple={multiple}
-                onChange={val => {
-                  const previousValue = props.value?.toString();
-                  const newValue = val.value?.toString();
-                  if (confirm && confirm[newValue] && previousValue !== newValue) {
-                    open(undefined, val);
-                    return;
-                  }
-                  this.handleChange.bind(this, onBlur, onChange)(val);
-                }}
-              />
-            )}
-          </Confirm>
+          <SelectControl
+            {...props}
+            clearable={allowClear}
+            multiple={multiple}
+            onChange={val => {
+              if (!confirm) {
+                this.handleChange(onBlur, onChange, val);
+                return;
+              }
+
+              // Support 'confirming' selections. This only works with
+              // `val` objects that use the new-style options format
+              const previousValue = props.value?.toString();
+              const newValue = val.value?.toString();
```

Comment:
could the `toString` here cause some weirdness when comparing values? Or is it necessary because of the constraints react-select has for string values.

Reply:
Yeah idk, this is copy over from what was implemented already. cc @mgaeta

Reply:
Sorry I don't remember why we had to add the `toString()` calls.

Reply:
Just for future reference, would be a good time for a comment :)